source/extensions/filters/http/ext_proc/matching_utils.cc

```diff
@@ -0,0 +1,101 @@
+#include "source/extensions/filters/http/ext_proc/matching_utils.h"
+
+#include <memory>
+
+#if defined(USE_CEL_PARSER)
+#include "parser/parser.h"
+#endif
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+absl::flat_hash_map<std::string, ExpressionManager::CelExpression>
+ExpressionManager::initExpressions(const Protobuf::RepeatedPtrField<std::string>& matchers) {
+  absl::flat_hash_map<std::string, ExpressionManager::CelExpression> expressions;
+#if defined(USE_CEL_PARSER)
+  for (const auto& matcher : matchers) {
+    if (expressions.contains(matcher)) {
+      continue;
+    }
+    auto parse_status = google::api::expr::parser::Parse(matcher);
+    if (!parse_status.ok()) {
+      throw EnvoyException("Unable to parse descriptor expression: " +
+                           parse_status.status().ToString());
+    }
+
+    Filters::Common::Expr::ExpressionPtr expression =
+        Extensions::Filters::Common::Expr::createExpression(builder_->builder(),
+                                                            parse_status.value().expr());
+
+    CelExpression cel_expr{parse_status.value(), std::move(expression)};
+    expressions.emplace(matcher, std::move(cel_expr));
+  }
+#else
+  ENVOY_LOG(warn, "CEL expression parsing is not available for use in this environment."
+                  " Attempted to parse " +
+                      std::to_string(matchers.size()) + " expressions");
+#endif
+  return expressions;
+}
+
+std::unique_ptr<ProtobufWkt::Struct>
```

Comment:
If you get rid of unique_ptr on the caller site, I think you can  just return `ProtobufWkt::Struct` as before.